### PR TITLE
add unit for StandardPipelineDataSourceConfiguration (#15710)

### LINKED ISSUE #15710: [OpenForce 2022] Add unit test for StandardPipelineDataSourceConfiguration
Hi community,

This issue is for [OpenForce 2022](https://github.com/apache/shardingsphere/discussions/12683#discussioncomment-2139441)

### Aim
Add unit test for `StandardPipelineDataSourceConfiguration` in `shardingsphere-pipeline-test` module.

### Basic Qualifications
- Java
- Maven
- Junit.Test

### PR DESCRIPTION
**Pull request by pparthenhs**
Fixes #15710

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StandardPipelineDataSourceConfigurationTest {
+
+    private StandardPipelineDataSourceConfiguration standardPipelineDataSourceConfiguration;
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java:41]: Could `standardPipelineDataSourceConfiguration` variable name be shorter? e.g. `dataSourceConfig`.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    @Before
+    public void setUp() throws SQLException {
+        standardPipelineDataSourceConfiguration = new StandardPipelineDataSourceConfiguration("jdbc:.*", "userName", "password");
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java:45]: Jdbc url could be more meaningful, e.g. `jdbc:mysql://127.0.0.1:3306/demo_ds?serverTimezone=UTC&useSSL=false`, you could refer to `StandardJdbcUrlParserTest` for more details.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        String actualDatabaseTypeName = actualDatabaseType.getName();
+        String expectedDatabaseTypeName = "SQL92";
+        assertThat(actualDatabaseTypeName, is(expectedDatabaseTypeName));
+    }
+
+    @Test
+    public void assertGetHikariConfigSuccess() {
+        HikariConfig actualHikariConfig = standardPipelineDataSourceConfiguration.getHikariConfig();
+        long actualActualHikariConfigConnectionTimeout = actualHikariConfig.getConnectionTimeout();
+        long expectedActualHikariConfigConnectionTimeout = new Long(30000);
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java:77]: `new Long(30000)` could be `30000`.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+               + "username: userName\n"
+               + "password: password\n";
+        assertThat(actualParameter, is(expectedParameter));
+    }
+
+    @Test
+    public void assertGetTypeSuccess() {
+        String actualType = standardPipelineDataSourceConfiguration.getType();
+        String expectedType = "JDBC";
+        assertThat(actualType, is(expectedType));
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java:94]: `expectedType` variable is not required, `is(expectedType)` could be `is("JDBC")`. And also other places.

I'm sorry, the code conduct doesn't fit for every case.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import java.sql.SQLException;
+import java.util.Properties;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class StandardPipelineDataSourceConfigurationTest {
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java:39]: Class could be `final`.

**Review by sandynz — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java (hunk 4 above)
Comment (pparthenhs) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java:94]: Dear @sandynz, 

Thank you for your comment and collaboration.

When we should use the `expectedXXX` pattern ??

I remain always at your disposal.

Best Regards, 
Panagiotis Parthenis

Comment (sandynz) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-spi/src/test/java/org/apache/shardingsphere/data/pipeline/api/datasource/config/StandardPipelineDataSourceConfigurationTest.java:94]: From guideline: `Pursue clean, simplified and extremely elegant codes`.
If the assign statement of `expectedXXX` is simple enough, then we could just put it into `is(...)`. e.g.
```
        assertThat(event.getType(), anyOf(is(Type.ADDED), is(Type.UPDATED)));
...
        assertThat(shardingItems.size(), is(1));
        assertThat(shardingItems.get(0), is(jobContext.getShardingItem()));
```

**Comment by pparthenhs** [comment]
Dear @sandynz, 

I am really thankful for your clarification and your comments.

I adapt the code according to your comments, so we could proceed if there are no more comments.

Best regards, 
Panagiotis Parthenis

**Comment by sandynz** [comment]
Hi @pparthenhs , I found there's failed CI, you could check the CI log for more details. One of the failure:
```
Error:  Failures: 
Error:    StandardPipelineDataSourceConfigurationTest.assertGetParameterSuccess:86 
Expected: is "jdbcUrl: jdbc:mysql://127.0.0.1:3306/demo_ds?serverTimezone=UTC&useSSL=false\nusername: userName\npassword: password\n"
     but: was "jdbcUrl: jdbc:mysql://127.0.0.1:3306/demo_ds?serverTimezone=UTC&useSSL=false\r\nusername: userName\r\npassword: password\r\n"
```

**Review by sandynz — APPROVED** [review]

**Comment by sandynz** [comment]
@pparthenhs Thanks for you PR.

**Comment by pparthenhs** [comment]
> Hi @pparthenhs , I found there's failed CI, you could check the CI log for more details. One of the failure:
> 
> ```
> Error:  Failures: 
> Error:    StandardPipelineDataSourceConfigurationTest.assertGetParameterSuccess:86 
> Expected: is "jdbcUrl: jdbc:mysql://127.0.0.1:3306/demo_ds?serverTimezone=UTC&useSSL=false\nusername: userName\npassword: password\n"
>      but: was "jdbcUrl: jdbc:mysql://127.0.0.1:3306/demo_ds?serverTimezone=UTC&useSSL=false\r\nusername: userName\r\npassword: password\r\n"
> ```

Hello @sandynzm, once more you are correct regarding test. After some investigation, this error is operation system-specific, so I adapt the code properly. 

I am happy to contribute !! :rocket:

**Comment by sandynz** [comment]
@pparthenhs Nice! Hope to see you next time.

And could you share why the error is operation system-specific and the related code? Maybe somebody is interested in it.

**Pull request merged**